_posts/2017-04-30-React-Native-for-iOS-devs.markdown

```diff
@@ -0,0 +1,671 @@
+
+## JS Tooling
+
+Facebook's tooling is 👍. They have an IDE-like text editor called Nuclide, which is built on GitHub's text editor Atom. If you're using the default setup for React Native then you'll probably have a good time with it.
+
+We opted for using TypeScript to provide a type system to our React Native - I covered why inside [Retrospective: Swift at Artsy][swift-at-artsy]. It's got substantially easier to use TypeScript with React Native since version 0.45 includes work from Artsy and [futurice][] making it possible via config settings alone.
+
+We choose TypeScript because of how well VS Code and TypeScript integrate. It provides an Xcode-like level of integration. Which, IMO, is a high baseline for expectations. It's definitely less polished, but it's Open Source - which is a very fair trade-off to me.
+
+The node community is great at automation, we have code formatters and language linters that will auto-fix your code as you press save. We have pre-commit, pre-push hooks that are set up automatically when you start using the project. It means you spend less time thinking about trivial details that add up. It's wonderful.
+
+We use the debugging tools built into Chrome for our React Native, instead of something like LLDB. It supports all of the kinds of runtime introspection you would expect from Xcode.
+
+# Testing
+
+Testing was never a priority in the iOS world. I feel like it's always getting better with each Xcode version, particularly the change in Xcode 9 which allows tests to run in another simulator without turning off your app.
+
+We really put a lot of time and patience into our native testing on iOS. Coming from that world, to the absolute ease of testing in the JavaScript world is pretty breath-taking. Check out my [coverage of Jest's features][artsy-jest].
+
+There are two ways to write tests for your React Native code: in process and out of process. E.g. from the JavaScript side, or in native side.
+
+-   JS side: Choice of many Open Source test runners, built with hundreds of contributors involved in multi-year test runners.
+
+-   Native side: Probably one or two people making XCTest, one person trying to get some improvements in Xcode each year. [Semi-closed source][xctest].
+
+We tried out a few native tests, but very quickly we stopped running them. Mainly as we were spending most of our time in a JavaScript environment, so tests would need to run in Xcode. As we don't need a Mac to run our tests, we can use linux CI servers and get 3-4 minute CI runs.
+
+# Deployment
```

Comment:
It could be worth mentioning the problem of the App Store review guideline issues that have surfaced, with Apple now saying RN updates to JS code aren't allowed. I haven't heard anything happening from the change, but people will likely ask.